src/include/daos_errno.h

```diff
@@ -212,39 +280,51 @@ D_FOREACH_ERR_RANGE(D_DEFINE_RANGE_ERRNO)
 
 /** Return a string associated with a registered gurt errno
  *
- * \param	rc[in]	The error code
+ * \param[in]	rc	The error code
  *
  * \return	String value for error code or DER_UNKNOWN
  */
 const char *d_errstr(int rc);
 
 /** Register error codes with gurt.  Use D_REGISTER_RANGE.
  *
- * \param	start[in]	Start of error range.  Actual errors start at
+ * \param[in]	start		Start of error range. Actual errors start at
  *				\p start + 1
- * \param	end[in]		End of range.  All error codes should be less
+ * \param[in]	end		End of range. All error codes should be less
  *				than \p end
- * \param	error_strings[in]	Array of strings.   Must be one per
- *					code in the range
+ * \param[in]	error_strings	Array of strings. Must be one per
+ *				code in the range
+ * \param[in]	strerror	Array of strings. Must be one per
+ *				code in the range
  *
  * \return	0 on success, otherwise error code
  */
 int d_errno_register_range(int start, int end,
-			   const char * const *error_strings);
+			   const char * const *error_strings,
+			   const char * const *strerror);
 
 /** De-register error codes with gurt.  Use D_DEREGISTER_RANGE.
  *
- * \param	start[in]	Start of error range
+ * \param[in]	start	Start of error range
  */
 void d_errno_deregister_range(int start);
 
+/** Return an error description string associated with a registered gurt errno.
+ *
+ * \param[in]	errnum	The error code
+ *
+ * \return		The error description string, or an "Unknown
+ *			error nnn" message if the error number is unknown.
+ */
+const char *d_errdesc(int errnum);
 
 /** @}
  */
 
 #ifndef DF_RC
 #define DF_RC "%s(%d)"
 #define DP_RC(rc) d_errstr(rc), rc
+#define DD_RC(rc) d_errdesc(rc), rc
```

Comment:
I think Ashley may have been asking for integration with the existing macro.   This is a little inconsistent with other DF (format) and DP (print) macros. so something like 

#define DF_RCV DF_RC ": '%s'"
#define DP_RCV(rc) DP_RC(rc), d_errdesc(rc)

would be more consistent.  I'm not sure exactly what Ashley had in mind but I think it might be to just add the description to the existing macro in which case you would do something like the above for the DF_RC macro.   Personally, I think that may be too verbose for most messages.

Reply:
Another option that would be consistent would be rename DD_RC to DP_RCV (verbose)

Reply:
I agree. It could be too verbose, especially for log files. I like your first suggestion. We could use it to report errors directly to the user such as dmg or pydaos.

Reply:
I think we should just have one DP_RC here, and it should use the full test string.  Having both DP_RC and DD_RC means a decision would need to be made in lots of invocation places which just wouldn't happen.